requirements_tools/check_requirements.py

```diff
@@ -394,7 +394,8 @@ def test_top_level_dependencies():
                 '\n'
                 'Usually this happens because you upgraded some other dependency, and now no longer require these.\n'  # noqa
                 "If that's the case, you should remove these from {pin}.\n"  # noqa
-                'Otherwise, if you *do* need these packages, then add them to {minimal}.\n'  # noqa
+                'Otherwise, if you *do* need these packages, which there is existing import of this package in your logic,'  # noqa
+                'then add them to {minimal}.\n'  # noqa
```

Comment:
this doesn't make grammatical sense -- and I think the original wording is clear, concise, and accurate

Reply:
Updated, just hope to make things being more specific on what we should actually do to know if a package is really needed.

It seems not making any sense with the package added manually, I was talking about a scenario that the dependencies could be automatically added by a software tool and the developer would need to verify if that package is really needed, or further provide update to other related dependencies would prevent the package being added.

Reply:
Thanks for reviewing @asottile !